jbi/actions/default.py

```diff
@@ -97,7 +99,15 @@ def __call__(self, context: ActionContext) -> ActionResult:
         responses = tuple()  # type: ignore
 
         for step in self.steps[context.operation]:
-            context, step_responses = step(context=context, **self.parameters)
+            try:
+                context, step_responses = step(context=context, **self.parameters)
+            except Exception:
+                if len(responses) > 0:
+                    # Count the number of workflows that produced at least request,
+                    # but could not complete entirely successfully.
+                    statsd.incr("jbi.bugzilla.partial.count")
+                raise
```

Comment:
I think this doesn't exactly cover what I was envisioning, though it could still be valuable to capture 🤔. 

For what I was thinking, say a user specifies these steps:
```python
"new": [
    "create_issue",
    "maybe_update_components",
],
```

The issue is created, but when we go to add the component that the user specified in `jira_components`, it isn't on the project for some reason, which hits this branch:

```python
if candidate_components:
    logger.warning(
        "Could not find components %s in project",
        candidate_components,
        extra=context.dict(),
    )
```

In this case, all of the steps in the workflow have run and no exceptions were raised, but the end result of the workflow is an issue that is missing some data the user expected to be there. The user might ask "why wasn't the component set? I have that in the config."

In this case, I'd consider the sync "incomplete", and increase `jbi.bugzilla.partial.count`. It's kind of a signal that while we didn't hit any exceptions while we were trying to complete the sync, we lost some data due to errors or misconfiguration along the way